cnd/src/network/comit.rs

```diff
@@ -97,143 +88,109 @@ impl Comit {
-        self.secret_hash
-            .register_addresses(peer.clone(), addresses.clone());
-        self.ethereum_identity
-            .register_addresses(peer.clone(), addresses.clone());
-        self.lightning_identity
-            .register_addresses(peer.clone(), addresses.clone());
-        self.finalize.register_addresses(peer.clone(), addresses);
-
-        self.ethereum_identity.send(
-            peer.clone(),
-            ethereum_identity::Message::new(swap_id, create_swap_params.ethereum_identity),
-        );
-        self.lightning_identity.send(
-            peer.clone(),
-            lightning_identity::Message::new(swap_id, create_swap_params.lightning_identity),
-        );
-
-        let seed = self.seed.derive_swap_seed(local_swap_id);
-        let secret_hash = SecretHash::new(seed.derive_secret());
-
-        self.secret_hashes.insert(swap_id, secret_hash);
-        self.secret_hash
-            .send(peer, secret_hash::Message::new(swap_id, secret_hash));
-
-        self.communication_state
-            .insert(swap_id, CommunicationState::default());
+    fn confirm(shared_swap_id: SharedSwapId, io: ReplySubstream<NegotiatedSubstream>) {
+        tokio::task::spawn(io.send(shared_swap_id));
     }
 
-    /// After announcement, confirm and then exchange information for the swap,
-    /// once done, go finalize
-    fn bob_communicate(
+    fn communicate(
         &mut self,
-        peer: libp2p::PeerId,
-        io: ReplySubstream<NegotiatedSubstream>,
         shared_swap_id: SharedSwapId,
-        create_swap_params: Herc20HalightBitcoinCreateSwapParams,
+        peer_id: libp2p::PeerId,
+        data: LocalData,
     ) {
-        // Confirm
-        tokio::task::spawn(io.send(shared_swap_id));
-
-        let addresses = self.announce.addresses_of_peer(&peer);
+        let addresses = self.announce.addresses_of_peer(&peer_id);
         self.secret_hash
-            .register_addresses(peer.clone(), addresses.clone());
+            .register_addresses(peer_id.clone(), addresses.clone());
         self.ethereum_identity
-            .register_addresses(peer.clone(), addresses.clone());
+            .register_addresses(peer_id.clone(), addresses.clone());
         self.lightning_identity
-            .register_addresses(peer.clone(), addresses.clone());
-        self.finalize.register_addresses(peer.clone(), addresses);
+            .register_addresses(peer_id.clone(), addresses.clone());
+        self.finalize.register_addresses(peer_id.clone(), addresses);
 
         // Communicate
-        self.ethereum_identity.send(
-            peer.clone(),
-            ethereum_identity::Message::new(shared_swap_id, create_swap_params.ethereum_identity),
-        );
-        self.lightning_identity.send(
-            peer,
-            lightning_identity::Message::new(shared_swap_id, create_swap_params.lightning_identity),
-        );
+        if let Some(ethereum_identity) = data.ethereum_identity {
```

Comment:
I think this trick is awesome 😃

Reply:
Sick huh, this is the key change that has made the **comit** module.